scripts/build.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+BASEDIR=$(pwd)
+git submodule update --init --recursive
+sudo apt-get install libgtest-dev && cd /usr/src/gtest && sudo cmake . && sudo make
```

Comment:
I don't think this is necessary for Monero, just requested. We can slim this down a good bit (notably the following apt) by removing optionals.

Reply:
that would be nice, i just copy-pasted this from the Monero readme lol

Reply:
This is the list of dependencies I used for building the project on Debian:

```
apt install cmake libzmq3-dev libboost-dev libboost-system-dev libboost-filesystem-dev libboost-thread-dev libboost-date-time-dev libboost-chrono-dev libboost-regex-dev libboost-serialization-dev libboost-program-options-dev libboost-locale-dev
```

May be sufficient for Ubuntu too since it's Debian-based.

Also, last time I checked Serai required Rust 1.60 stable.

Reply:
Thanks for chiming in with that :)

I'm actually not sure what version of Rust we require + why. It may be good to go through each crate, working that, and adding `msrv`s accordingly, working on finding earlier versions which are trivial to move to in order to bring wide support.

Reply:
I checked out the latest commit and there were 458 new crates to download :exploding_head: 
- `librocksdb-sys` requires `apt install clang`
- `ethereum-serai` requires solc binary (I think it can be installed from PPA on Ubuntu)
- `serai-runtime` requires rust-nightly
- `monero-serai` fails with ```no method named `len` found for type `usize` in the current scope``` (I'm not sure if it has anything to do with rust version)
- Apparently the rest of dependency tree still builds with 1.60, though I didn't get to the end of it

Rust 1.60 requirement probably comes from namespaced features in [some manifests](https://github.com/serai-dex/serai/blob/develop/crypto/transcript/Cargo.toml#L19) - that was the error I encountered when I tried to build project with 1.56 initially.

Reply:
The new crates would be the mess that is Substrate.

solc should be from solc-select, which is a pip3.

monero-serai is absolutely my fault and dates back to when I cleaned up CLSAG a bit due to cargo fmt's opinion on it.

Anything wasm under Substrate will require nightly :/

Thanks for clarifying there. I forgot Rust only recently got sane feature naming...